src/write.rs

```diff
@@ -476,7 +483,177 @@ where
                 Absorb::absorb_second(w_inner, op, &*r_handle);
             }
         } else {
-            self.oplog.extend(ops);
+            // Only try to compress if it is enabled, else use efficient fallback.
+            if T::MAX_COMPRESS_RANGE > 0 {
+                // Compress oplog by rev-iterating all ops appended since the last publish
+                // while attempting to combine them with the next op,
+                // cut short when an attempt fails due to encountering a dependency (e.g. clear then set).
+
+                // used to avoid walking more of the oplog than necessary.
+                let mut rev_dirty_range = 0usize..0;
+                for mut next in ops {
```

Comment:
I wonder whether it might be more efficient to _first_ compress `ops` on its own, and _then_ compress the resulting list of operations into `self.ops`. What do you think?

Reply:
The compression working back-to-front instead of front-to-back means we would need to `collect` them first, meaning we would have to (de-)allocate every single call, even if only inserting a single op.